thousandeyes/data_source_thousandeyes_integration.go

```diff
@@ -95,6 +95,10 @@ func dataSourceThousandeyesIntegrationRead(d *schema.ResourceData, meta interfac
 	if err != nil {
 		return err
 	}
+	err = d.Set("integration_type", found.IntegrationType)
+	if err != nil {
+		return err
+	}
```

Comment:
I would suggest we remove all of these `err = d.Set("integration_X", found.IntegrationName)` and replace them all with:
```go
	err = ResourceRead(d, &found)
	if err != nil {
		return err
	}
```

So it would look like this:
```go
(...)
	if found == (thousandeyes.Integration{}) {
		return fmt.Errorf("unable to locate any integration by name: %s", searchName)
	}

	d.SetId(*found.IntegrationID)
	err = ResourceRead(d, &found)
	if err != nil {
		return err
	}
```

That way we're simplifying this and reducing the amount of conditional statements. How does this sound? 
I tested this and it worked.

Reply:
In addition (just thought of this) we can remove the `auth_token` from the schema, as it doesn't seem to be used, the API doesn't return it, and even if it did we wouldn't want to store the token in the state, I think.